agent/api/task_unix.go

```diff
@@ -15,10 +15,48 @@
 
 package api
 
+import (
+	"github.com/aws/amazon-ecs-agent/agent/resources/cgroup"
+	docker "github.com/fsouza/go-dockerclient"
+
+	"github.com/pkg/errors"
+)
+
 const (
 	portBindingHostIP = "0.0.0.0"
 )
 
 func (task *Task) adjustForPlatform() {}
 
 func getCanonicalPath(path string) string { return path }
+
+// GetCgroupSpec fetches the task cgroup spec
+func (task *Task) GetCgroupSpec() (cgroup.Spec, error) {
+	task.cgroupSpecLock.RLock()
+	defer task.cgroupSpecLock.RUnlock()
+
+	if task.CgroupSpec == nil {
+		return cgroup.Spec{}, errors.New("task cgroup: missing spec")
+	}
+
+	return *task.CgroupSpec, nil
+}
+
+// updateHostConfigWithCgroupParent sets the cgroup parent for containers
+func (task *Task) updateHostConfigWithCgroupParent(hostConfig *docker.HostConfig) error {
+	// Get cgroup spec
+	cgroupSpec, err := task.GetCgroupSpec()
+	if err != nil {
+		return errors.Wrapf(err, "task set cgroup parent: unable to get valid cgroup spec")
+	}
+
+	// Check for empty cgroup root
+	if cgroupSpec.Root == "" {
+		return errors.New("task set cgroup parent: empty cgroup root")
+	}
+
+	// Set cgroup parent
+	hostConfig.CgroupParent = cgroupSpec.Root
```

Comment:
This is still odd. You shouldn't be modifying external configuration from within a function on an unrelated type.